src/open_feature_api.py

```diff
@@ -0,0 +1,35 @@
+from src.open_feature_client import OpenFeatureClient
+from src.provider.provider import AbstractProvider
+
+global provider
```

Comment:
I think you don't need the `global` declaration here. The `global` declaration is only required within the scope of another object in the module, e.g. a function / class method. 

So, we should change this to just `provider = None`.